crates/ra_ide_api/src/diagnostics.rs

```diff
@@ -218,6 +231,43 @@ mod tests {
         assert_eq!(diagnostics.len(), 0);
     }
 
+    #[test]
+    fn test_wrap_return_type() {
+        let before = r#"
+            enum Result<T, E> { Ok(T), Err(E) }
```

Comment:
My initial confusion around matching types came from the fact that the inferred return type for my test function was Unknown. I initially assumed that this was because the function body didn't type-check, but it was actually because the db didn't know about std::result::Result. So I added my own types here. How should I make the db aware of the std types here?

Reply:
The unit tests are standalone, because loading all of std for each test would be much too expensive. So this is the right approach (except that we should probably get `Result` by its full path in std).

Reply:
(It would maybe be nice to have a standard set of mock std types for the unit tests, though.)

Reply:
Ok, good to know.